### PR TITLE
Feat/create user authentication

### PR DESCRIPTION
**Pull request by mllrdev-glenn-henares**

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -3,7 +3,7 @@
   <head>
     <meta charset="utf-8" />
     <title>Ionic App</title>
-
+    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.1.0/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-KyZXEAg3QhqLMpG8r+8fhAXLRk2vvoC2f3B09zVXn8CA5QIVfZOJ3BCsw2P0p/We" crossorigin="anonymous">
```
Comment (cerinoligutom) [inline public/index.html]: Do you even need this? If yes, use a solution that doesn't use a CDN.

File: src/components/Toolbar.vue [hunk 2]
```diff
@@ -0,0 +1,41 @@
+<template>
+  <ion-header :translucent="true">
+    <ion-toolbar>
+      <div id="title-bar" >
```
Comment (cerinoligutom) [inline src/components/Toolbar.vue]:
```suggestion
      <div class="title-bar" >
```

Just some semantics, only use `id` if it's really unique throughout the entire app AND you want to access it in javascript. If the former doesn't apply, opt for using css classes.

File: src/components/Toolbar.vue [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import { defineComponent } from "vue";
+
+export default defineComponent({
+  name: "Toolbar",
+  components: {
+    IonTitle,
+    IonHeader,
+    IonToolbar
+  },
+  props: ['titleText'],
```
Comment (cerinoligutom) [inline src/components/Toolbar.vue]:
```suggestion
  props: {
    titleText: {
      type: String,
      required: true
    },
  },
```

Always use the full form. See [style guide for prop definitions](https://v3.vuejs.org/style-guide/#prop-definitions-essential).

File: src/core/enums/route-names.enum.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+export enum RouteNames {
```
Comment (cerinoligutom) [inline src/core/enums/route-names.enum.ts]:
```suggestion
export enum RouteName {
```

enums should be in singular form unless the values are BIT

File: src/core/interfaces/user.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
+interface IName {
+    firstName: string;
+    lastName: string;
+}
+
+interface IUser extends IName {
+    email: string;
+    password: string;
+}
```
Comment (cerinoligutom) [inline src/core/interfaces/user.ts]: What benefits do you see in separating these interfaces? I believe they can be merged, no?

File: src/core/models/api-urls.ts [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import axios from "axios"
+
+const apiUrl = 'http://localhost:8081/api'
+const apiLoginUrl = apiUrl + '/user/login'
+const apiSignupUrl = apiUrl + '/user/sign-up'
+
+export default { apiUrl, apiLoginUrl, apiSignupUrl}
```
Comment (cerinoligutom) [inline src/core/models/api-urls.ts]: This isn't a model (based on directory path and content of this file) so this does not belong here.

File: src/core/services/api.service.ts [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+                router.push("/home")
+            }
+            else if (!response.data.isSuccessful) {
+                router.push("/login")
+            }
+        })
+        .catch(function (error) {
+            console.log("Error on Authentication");
+        })
+}
```
Comment (cerinoligutom) [inline src/core/services/api.service.ts]: Wrong api service.

File: src/pages/Register.vue [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+  name: "Home",
+  components: {
+    IonContent,
+    IonPage,
+    IonInput,
+    Toolbar,
+  },
+   setup() {
+    const router = useRouter();
+    return { router };
```
Comment (cerinoligutom) [inline src/pages/Register.vue]: Isn't this unnecessary?

File: src/pages/Register.vue [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+}
+#login {
+  justify-content: center;
+  display: flex !important;
+  flex-direction: column;
+  align-items: center;
+  padding: 10px;
+}
+ion-item {
+  width: 5in;
```
Comment (cerinoligutom) [inline src/pages/Register.vue]: Don't use the `in` unit for styling an app.  Usually that unit is used when you're trying to print a web page and have constraints for the paper size.

**Review by cerinoligutom — CHANGES_REQUESTED** [review]

**Review by mllrdev-carlo-medel — APPROVED** [review]

**Review by mllrdev-carlo-medel — APPROVED** [review]

**Pull request merged**